### PR TITLE
Refinements of helm resources and documents

### LINKED ISSUE #139: Promote Helm resources as the primary and single source of Kubernetes resource installation
**Description:**
Promote Helm resources for WSO2 Identity and Access Management deployment patterns as the primary and single source of Kubernetes resource installation. Hence, it has been suggested to deprecate the Kubernetes resources for WSO2 Identity and Access Management deployment patterns in the future.

**Affected Product Version:**
Kubernetes and Helm resources for WSO2 Identity Server v5.8.x and below

### LINKED ISSUE #155: Harden security
**Description:**
Fine-tune config parameters for secure deployments on production setups. 

**Affected Product Version:**
 Helm resources for WSO2 Identity Server v5.8.x and below

### LINKED ISSUE #156: Add datasource configurations to values.yaml
**Description:**
Add configurations for MySQL datasources as follows to the values.yaml file in helm resources for all deployment patterns.

```
mysql:
    enabled: true
    host: wso2is-rdbms-service-mysql
    username: wso2carbon
    password: wso2carbon
    driverClass: com.mysql.jdbc.Driver
    validationQuery: "SELECT 1"
```

These values can be referred in the ConfigMap templates as follows:
```
{{- $config_dburl := printf "%s%s%s" "jdbc:mysql://" .Values.wso2.mysql.host ":3306/WSO2EI_INTEGRATOR_CONFIG_GOV_DB?autoReconnect=true&amp;useSSL=false" }}
{{- $user_dburl := printf "%s%s%s" "jdbc:mysql://" .Values.wso2.mysql.host ":3306/WSO2EI_USER_DB?autoReconnect=true&amp;useSSL=false" }}
{{- $db_driver_class_name := .Values.wso2.mysql.driverClass }}
{{- $db_validation_query := .Values.wso2.mysql.validationQuery }}
{{- $db_username := .Values.wso2.mysql.username }}
{{- $db_password := .Values.wso2.mysql.password }}
```

Additionally, the database host can be parameterized in the deployment template initContainers as follows:
```
command: ['sh', '-c', 'echo -e "Checking for the availability of MySQL Server deployment"; while ! nc -z {{ default "wso2ei-rdbms-service-mysql" .Values.wso2.mysql.host }} 3306; do sleep 1; printf "-"; done; echo -e "  >> MySQL Server has started";']
```

**Affected Product Version:**
 Helm resources for WSO2 Identity Server v5.8.x and below

### PR DESCRIPTION
**Pull request by Savidude**

### CODE REVIEW

File: advanced/helm/is-pattern-2/values.yaml [hunk 1]
```diff
@@ -33,9 +40,15 @@ wso2:
         initialDelaySeconds: 250
         periodSeconds: 10
       resources:
+        # These are the minimum resource recommendations for running WSO2 Enterprise Integrator server profiles
```
Comment (chirangaalwis) [inline advanced/helm/is-pattern-2/values.yaml]: @Savidude these comments have to change such that they are applicable to WSO2 Identity Server.

Comment (Savidude) [reply advanced/helm/is-pattern-2/values.yaml]: Fixed https://github.com/wso2/kubernetes-is/pull/159/commits/e5afef47cc931b20789608fccfb3be0b0513fa3f

File: CHANGELOG.md [hunk 2]
```diff
@@ -19,3 +19,24 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - Integrate support in Helm resources for users with and without WSO2 subscriptions
 
 [v5.8.0.2]: https://github.com/wso2/kubernetes-is/compare/v5.8.0.1...v5.8.0.2
+
+## [V5.8.0.3] - 2019-08-15
```
Comment (chirangaalwis) [inline CHANGELOG.md]: @Savidude let's keep this as `Unreleased` as in the example. We can set the date to the specific release date when we merge this branch with the `master` branch.

Comment (Savidude) [reply CHANGELOG.md]: Changed https://github.com/wso2/kubernetes-is/pull/159/commits/2d0ceb62519ef48cbdcb9f3be7ebcc1b1f911332

**Pull request merged**